src/storage/overlay_root.rs

```diff
@@ -175,7 +176,7 @@ impl StorageEngine {
                     if overlay.is_empty() && can_add_by_hash {
                         if let Some(hash) = pointer.rlp().as_hash() {
                             // No overlay, just add the pointer by hash
-                            root_builder.add_branch(path, hash, true);
+                            root_builder.add_branch(path.try_into().unwrap(), hash, true);
```

Comment:
Should this properly handle the error or do we not expect the `unwrap` to be hit?